openhcl/underhill_attestation/src/igvm_attest/mod.rs

```diff
@@ -181,10 +181,19 @@ fn create_request(
 /// Get the expected size of the given report type.
 fn get_report_size(report_type: IgvmAttestReportType) -> Result<usize, Error> {
     let size = match report_type {
-        IgvmAttestReportType::VBS_VM_REPORT => protocol::igvm_attest::get::VBS_VM_REPORT_SIZE,
-        IgvmAttestReportType::SNP_VM_REPORT => protocol::igvm_attest::get::SNP_VM_REPORT_SIZE,
-        IgvmAttestReportType::TDX_VM_REPORT => protocol::igvm_attest::get::TDX_VM_REPORT_SIZE,
-        _ => Err(Error::InvalidAttestationReportType)?,
+        IgvmAttestReportType::VBS_VM_REPORT => {
+            openhcl_attestation_protocol::igvm_attest::get::VBS_VM_REPORT_SIZE
+        }
+        IgvmAttestReportType::SNP_VM_REPORT => {
+            openhcl_attestation_protocol::igvm_attest::get::SNP_VM_REPORT_SIZE
+        }
+        IgvmAttestReportType::TDX_VM_REPORT => {
+            openhcl_attestation_protocol::igvm_attest::get::TDX_VM_REPORT_SIZE
+        }
+        IgvmAttestReportType::TVM_REPORT => {
+            openhcl_attestation_protocol::igvm_attest::get::TVM_REPORT_SIZE
+        }
+        ty => Err(Error::InvalidAttestationReportType(ty.0))?,
```

Comment:
this seems like the sort of error you can eliminate at compile time, simply by defining a Rust-enum which maps back to the underying protocol type. You would store the Rust num in `self.report_type`, and then this function would avoid the branch